### PR TITLE
feat: rate limiting for the visRawReader

### PR DESCRIPTION
**Pull request by jrs65**
This adds the option for limiting the read rate for visRawReader to try
and reduce issues with gossec over NFS. It also adds the ability to
create VisRaw style files from Python space for testing.

### CODE REVIEW

File: lib/stages/visRawReader.cpp [hunk 1]
```diff
@@ -296,5 +308,15 @@ void visRawReader::main_thread() {
...
         ind++;
+
+        // Get the end time for the loop and sleep for long enough to satisfy
+        // the max rate
+        end_time = current_time();
+        double sleep_time = min_read_time - (end_time - start_time);
+        DEBUG("Sleep time {}", sleep_time);
+        if (sleep_time > 0) {
+            auto ts = double_to_ts(sleep_time);
+            nanosleep(&ts, NULL);
```
Comment (nritsche) [inline lib/stages/visRawReader.cpp]: I guess the sleep times end up so short, that `usleep` wouldn't be precise enough here?

File: python/kotekan/visbuffer.py [hunk 2]
```diff
@@ -21,6 +21,13 @@ class time_spec(ctypes.Structure):
 
     _fields_ = [("tv", ctypes.c_int64), ("tv_nsec", ctypes.c_uint64)]
 
+    @classmethod
+    def from_float(cls, v):
```
Comment (nritsche) [inline python/kotekan/visbuffer.py]: add documentation
e.g.
```
Create a time_spec from a float.

Parameters
----------------
v : float
    The interval in seconds.
```

File: python/kotekan/visbuffer.py [hunk 3]
```diff
@@ -365,6 +374,160 @@ def __init__(self, filename, mmap=False):
...
+    """
+
+    nprod = ninput * (ninput + 1) // 2
+    nev = 4
+
+    time = [{"ctime": (10.0 * i), "fpga_count": i} for i in range(ntime)]
+
+    freq = [{"centre": (800 - i * 10.0), "width": 10.0} for i in range(10)]
+
+    input_ = [(i, f"test{i:04}") for i in range(ninput)]
```
Comment (nritsche) [inline python/kotekan/visbuffer.py]: This only works for python >= 3.7.
```suggestion
    input_ = [(i, "test{:04}".format(i)) for i in range(ninput)]
```

File: lib/stages/visRawReader.cpp (hunk 1 above)
Comment (jrs65) [reply lib/stages/visRawReader.cpp]: Nah, I don't think they're that short, but as it was easy enough for me to get a timespec I figured why not.

**Pull request merged**